gstprovider/gstthread.cpp

```diff
@@ -189,7 +189,7 @@ class GstSession
         int need_maj = 1;
         int need_min = 4;
         int need_mic = 0;
-        if(compare_gst_version(major, minor, micro, need_maj, need_min, need_mic) < 0)
+        if(compare_gst_version(int(major), int(minor), int(micro), need_maj, need_min, need_mic) < 0)
```

Comment:
why not just fix compare_gst_version to accept unsigned vairants instead

Reply:
fixed